src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -218,6 +215,24 @@ def start():
     _session.start_time = datetime.datetime.now()
 
 
+@decorators.suppress_all_exceptions(raise_in_diagnostics=True)
+def flush():
+    global _session  # pylint: disable=global-statement
+    # flush out current information
+    _session.end_time = datetime.datetime.now()
+
+    payload = _session.generate_payload()
+    if payload:
+        import subprocess
+        subprocess.Popen([sys.executable, os.path.realpath(telemetry_core.__file__), payload])
+
+    # reset fields
+    new_session = TelemetrySession()
+    new_session.correlation_id = _session.correlation_id
+    new_session.application = _session.application
```

Comment:
Shouldn't this just be done in the constructor?
`TelemetrySession()` should just set up everything ready to go.
Kind of like https://softwareengineering.stackexchange.com/questions/334970/are-init-methods-a-code-smell?